test/profile/test_profile.py

```diff
@@ -72,6 +73,33 @@ def test_default_profiling(self):
             visualize.profiles(result, profile_list_ids=i_run)
             visualize.profile_cis(result, profile_list=i_run)
 
+    def test_engine_profiling(self):
+        # loop over all possible engines
+        # engine=None will be used for comparison
+        engines = [None,
+                   pypesto.engine.SingleCoreEngine(),
+                   pypesto.engine.MultiProcessEngine(),
+                   pypesto.engine.MultiThreadEngine()
+                   ]
+        for engine in engines:
+            # run profiling, profile results get appended
+            # in self.result.profile_result
+            profile.parameter_profile(
+                problem=self.problem,
+                result=self.result,
+                optimizer=self.optimizer,
+                next_guess_method='fixed_step',
+                engine=engine,)
+
+        # check results
+        for count, _engine in enumerate(engines):
+            for j in range(len(self.result.profile_result.list[0])):
```

Comment:
```suggestion
            for j in range(1, len(self.result.profile_result.list[0])):
```

I guess the first can be skipped

Reply:
why can the first be skipped? i think the first count can be skipped not j.

Reply:
Oops, meant `count`, added a new suggestion.